charts/ocis/templates/graph/deployment.yaml

```
@@ -251,7 +251,7 @@ spec:
           emptyDir: {}
           {{ end }}
         - name: ldap-ca
-          {{ if not .Values.features.externalUserManagement.enabled }}
+          {{ if or (not .Values.features.externalUserManagement.enabled) (not .Values.features.externalUserManagement.ldap.certTrusted) }}
```

Definitely the condition is missing here. I'm just questioning already the existing `or` condition especially when comparing that to the same thing for the messaging system: https://github.com/owncloud/ocis-charts/blob/b05a650a97b821e3abe6e347b4eafe3d637d290e/charts/ocis/templates/graph/deployment.yaml#L247

So to me the condition need to be the same as the one for the messaging system, so

```suggestion
          {{ if and (.Values.features.externalUserManagement.enabled) (not .Values.features.externalUserManagement.ldap.certTrusted) }}
```

@wkloucek what do you think?

---

I basically copied over the same condition as in the users Deployment. But I'm not really sure if `and` or `or` should be used in this case.

https://github.com/owncloud/ocis-charts/blob/b05a650a97b821e3abe6e347b4eafe3d637d290e/charts/ocis/templates/users/deployment.yaml#L179

---

groups, graph and users services must have the same condition because this are the only services that connect to the LDAP server and all LDAP specifics apply to all three of them.

groups and users already have the same condition (but some different whitespaces, so a search doesn't find both occurrences).

The logic is:

If we're using the builtin user management / LDAP (`not .Values.features.externalUserManagement.enabled`), we need to use the LDAP CA of the builtin LDAP / IDM service, because this it speaks LDAPS only and we don't want to fallback to insecure connections / skipping certificate validation and we have the CA in place anyways.

If an external LDAP is used, this applies:

https://github.com/owncloud/ocis-charts/blob/b05a650a97b821e3abe6e347b4eafe3d637d290e/charts/ocis/values.yaml#L180-L182


@d7oc if you translate the conditions literally, you'd end up with:

```
{{ if or (not .Values.features.externalUserManagement.enabled) (and .Values.features.externalUserManagement.enabled (not .Values.features.externalUserManagement.ldap.certTrusted)) }}
```

but you can reduce it to the current condition we have, because if you set `.Values.features.externalUserManagement.ldap.certTrusted` to `true` while you're using the builtin user management, you don't change anything.

---

I also fixed the differing whitespace in https://github.com/owncloud/ocis-charts/pull/828/commits/909f7f47af21ff149d945574e12a590acb96c9bf

---

Ok fine for me.